crates/serialize_hierarchy_derive/src/serialize_hierarchy.rs

```diff
@@ -23,32 +23,32 @@ pub fn process_serialize_hierarchy_implementation(
 
     let expanded = quote! {
         impl #impl_generics serialize_hierarchy::SerializeHierarchy for #name #ty_generics #where_clause {
-            fn serialize_hierarchy(&self, field_path: &str) -> anyhow::Result<serde_json::Value> {
-                use anyhow::Context;
+            fn serialize_hierarchy(&self, field_path: &str) -> color_eyre::eyre::Result<serde_json::Value> {
+                use color_eyre::eyre::WrapErr;
                 let split = field_path.split_once(".");
                 match split {
                     Some((field_name, suffix)) => match field_name {
                         #(#path_serialization,)*
-                        _ => anyhow::bail!("No such field in type: `{}`", field_path),
+                        _ => color_eyre::eyre::bail!("no such field in type: `{}`", field_path),
                     },
                     None => match field_path {
                         #(#serde_serialization,)*
-                        _ => anyhow::bail!("No such field in type: `{}`", field_path),
+                        _ => color_eyre::eyre::bail!("no such field in type: `{}`", field_path),
```

Comment:
It needs to stay because the injected serializations may use `wrap_err()`.

Reply:
I know, that's not what I meant. Why not also `use` `bail`?

Reply:
`WrapErr` is only `use`d because we need to use it to get access to the trait. All other identifiers are not used and accessed by fully qualified paths because of [macro hygiene](https://danielkeep.github.io/tlborm/book/mbe-min-hygiene.html). I would like to keep it like that.

Reply:
Interesting, I didn't know that.
Correct me if I'm wrong, but the way I read that is that the compiler basically does the hygiene/separation for you and you don't have to worry about polluting the user's namespace with `use`s?

Reply:
I think it does not do anything. This is the reason why we have macros. They inject tokens into the tokenstream that the compiler consumes. If we would use something globally, it will be used in that file. Therefore `WrapErr` is used within the scope to ensure hygiene and don't pollute the caller's scope. Using fully qualified paths is also hygiene.